typescript/cli/src/deploy/utils.ts

```diff
@@ -41,28 +40,53 @@ export async function runPreflightChecksForChains({
 }: {
   context: WriteCommandContext;
   chains: ChainName[];
-  minGas: string;
+  minGas: typeof MINIMUM_WARP_DEPLOY_GAS;
   // Chains for which to assert a native balance
   // Defaults to all chains if not specified
   chainsToGasCheck?: ChainName[];
 }) {
   log('Running pre-flight checks for chains...');
-  const { multiProvider, skipConfirmation } = context;
+  const {
+    multiProvider,
+    skipConfirmation,
+    multiProtocolProvider,
+    multiProtocolSigner,
+  } = context;
+
+  assert(multiProtocolSigner, 'multiProtocolSigner not defined');
 
   if (!chains?.length) throw new Error('Empty chain selection');
   for (const chain of chains) {
     const metadata = multiProvider.tryGetChainMetadata(chain);
     if (!metadata) throw new Error(`No chain config found for ${chain}`);
-    if (metadata.protocol !== ProtocolType.Ethereum)
-      throw new Error('Only Ethereum chains are supported for now');
-    const signer = multiProvider.getSigner(chain);
-    assertSigner(signer);
+
+    let signer: TypedSigner;
+
+    switch (metadata.protocol) {
+      case ProtocolType.Ethereum:
+        signer = multiProtocolSigner.getEVMSigner(chain);
+        break;
+      case ProtocolType.CosmosNative:
+        signer = multiProtocolSigner.getCosmosNativeSigner(chain);
+        break;
+      default:
+        throw new Error(
+          'Only Ethereum and Cosmos Native chains are supported for now',
+        );
+    }
```

Comment:
This whole block rather belongs to the `MultiProtocolSigner[Manager]`. It is concerned with getting the signer for the appropriate chain protocol, or even just checking if the signer is configured. The multi-protocol signer knows the chain metadata, so all this logic can be moved.